File: android/src/ti/airship/components/InAppAutomationProxy.java
/* Copyright Airship and Contributors */

package ti.airship.components;

import com.urbanairship.automation.InAppAutomation;

import org.appcelerator.kroll.KrollProxy;
import org.appcelerator.kroll.annotations.Kroll;

import java.util.concurrent.TimeUnit;

import ti.airship.utils.PluginLogger;

@Kroll.proxy
public class InAppAutomationProxy extends KrollProxy {

    @Kroll.method
    @Kroll.getProperty
    public boolean getIsPaused() {
        return InAppAutomation.shared().isPaused();
    }

    @Kroll.method
    @Kroll.setProperty
    public void setIsPaused(boolean paused) {
        PluginLogger.logCall("setIsPaused", paused);
        InAppAutomation.shared().setPaused(paused);
    }

    @Kroll.method
    @Kroll.getProperty
    public long getDisplayIntervalMilliseconds() {
        return InAppAutomation.shared().getInAppMessageManager().getDisplayInterval();
    }

    @Kroll.method
    @Kroll.setProperty
    public void setDisplayIntervalMilliseconds(long time) {
        PluginLogger.logCall("setDisplayIntervalMilliseconds", time);
        InAppAutomation.shared().getInAppMessageManager().setDisplayInterval(time, TimeUnit.MILLISECONDS);
    }
}

File: android/src/ti/airship/components/InAppAutomationProxy.kt
/* Copyright Airship and Contributors */
package ti.airship.components

import com.urbanairship.automation.InAppAutomation
import org.appcelerator.kroll.KrollProxy
import org.appcelerator.kroll.annotations.Kroll.*
import ti.airship.utils.PluginLogger
import java.util.concurrent.TimeUnit

@proxy
class InAppAutomationProxy : KrollProxy() {
    @get:getProperty
    @set:setProperty
    var isPaused: Boolean
        get() = InAppAutomation.shared().isPaused
        set(paused) {
            PluginLogger.logCall("setIsPaused", paused)
            InAppAutomation.shared().isPaused = paused
        }

    @get:getProperty
    @set:setProperty
    var displayIntervalMilliseconds: Long
        get() = InAppAutomation.shared().inAppMessageManager.displayInterval
        set(time) {
            PluginLogger.logCall("setDisplayIntervalMilliseconds", time)
            InAppAutomation.shared().inAppMessageManager.setDisplayInterval(
                time,
                TimeUnit.MILLISECONDS
            )
        }
}